### ISSUE TITLE
alternative repr that does not look like it can be evaluated

### ISSUE THREAD

**Issue by wbolster** [python-attrs/attrs#41]
the `repr()` generated by `attrs` looks something like this by default:

```
MyClass(a=1, b=2)
```

this is fine in simple cases. however, when using `init=False`, and implementing my own `__init__()`, this `repr()` is misleading since it looks like code that can be evaluated, while actually it is not (and if it is, it's not an intended usage pattern anyway). what about having something like this instead for those cases:

```
<MyClass a=1 b=2>
```

...which is still very useful (e.g. for debugging), and it resembles a bit what python produces for classes that do not implement a custom `__repr__()`:

```
>>> x = object()
>>> x
<object object at 0x7f6cda0c45e0>
```

obviously, this custom `repr()` behaviour would require explicit enabling. a flag to `attr.s()`, e.g. `attr.s(use_eval_repr=False)` (please come up with a better name), would make sense to me.

what do you think?

**Comment by wbolster** [comment]
btw, i can take a stab at a pr if you think this feature is worth adding. is seems that passing a flag to `_add_repr()` and using a slightly different format string depending on that flag is pretty much all that is needed.

**Comment by hynek** [comment]
I agree but I’m not sure if it’s worth the complexity and confusion.  Just because the `__init__` doesn’t come from us, doesn’t mean it’s not evaluable and users might get confused by the changing behavior.

**Comment by wbolster** [comment]
that's why i suggested to have another flag to enable this "custom repr". it would be an independent flag (not dependent on `init=False` in any way) so that it's just an additional opt-in feature.

**Comment by Julian** [comment]
I think I have an alternate solution that might satisfy this too. Adding some thoughts to a new issue (will link it in a sec).

**Comment by hynek** [comment]
Solved by #568!

**Comment by wbolster** [comment]
not sure that's correct, @hynek. i opened this ticket about custom `repr` for classes,

but #568 (also written by me) is about attribute-level custom `repr`.

see the examples in the description of this issue; i don't think that can be achieved with #568.

 (tentatively reopening the ticket, feel free to close if things turn out to be different from what i wrote.)

**Comment by Julian** [comment]
IIRC this ticket was about "Make `attr.s(repr=`, rather than `attr.ib(repr=` which is #568, *also* take a callable which collects and assembles the final one (and then secondly possibly implement a concrete callable implementation for bracket-looking-repr or soemthing).

**Comment by wbolster** [comment]
exactly, @Julian. in #568 i added a `repr=...` with a custom callable for `attr.ib()`.

in this (much older!) issue i didn't suggest a specific API, but i agree using `repr=...` with a custom callable, but for `attr.s()` seems the logical choice.

**Comment by JoshLipschultz** [comment]
Just wanted to point out that this would be especially useful in conjunction with https://etils.readthedocs.io/en/stable/api/epy/pretty_repr_top_level.html, so +1 on this feature request.
